src/OpenTelemetry/Metrics/Exemplar/IExemplarFilter.cs

```diff
@@ -0,0 +1,62 @@
+// <copyright file="IExemplarFilter.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Metrics;
+
+/// <summary>
+/// The interface defining Exemplar Filter.
+/// </summary>
+internal interface IExemplarFilter
+{
+    /// <summary>
+    /// Determines if a given measurement is eligible for being
+    /// considered for becoming Exemplar.
+    /// </summary>
+    /// <param name="value">The value of the measurement.</param>
+    /// <param name="tags">The complete set of tags provided with the measurement.</param>
+    /// <returns>
+    /// Returns
+    /// <c>true</c> to indicate this measurement is eligible to become Exemplar
+    /// and will be given to an ExemplarReservoir.
+    /// Reservoir may further sample, so a true here does not mean that this
+    /// measurement will become an exemplar, it just means it'll be
+    /// eligible for being Exemplar.
+    /// <c>false</c> to indicate this measurement is not eligible to become Exemplar
+    /// and will not be given to the ExemplarReservoir.
+    /// </returns>
+    bool ShouldSample(long value, ReadOnlySpan<KeyValuePair<string, object>> tags);
```

Comment:
Do we need an explicit parameter for timestamp? When would this method be called with respect to when a metric is recorded?

Reply:
spec says timestamp is needed. but we do not allow user to provide timestamp for metric APIs, and is always assumed to be the current time.. so the Filter can also take current timestamp, without we providing it. (have to check if this is bad for perf.. )

(Similar with context. we dont pass it. its implicit.

Reply:
No concerns with thread preemption here, right?